verification/spec/relayer/ConnectionHandlers.tla

```diff
@@ -0,0 +1,168 @@
+ connection datagrams
+ ***************************************************************************)
+
+EXTENDS Naturals, FiniteSets
+
+ConnectionIDs == {"connAtoB", "connBtoA"}
+nullConnectionID == "none"
+
+ConnectionStates == {"UNINIT", "INIT", "TRYOPEN", "OPEN"}
+
+(***************************************************************************
+ Connection helper operators
+ ***************************************************************************)
+
+\* get the connection ID of the connection end at chainID
+GetConnectionID(chainID) ==
+    IF chainID = "chainA"
+    THEN "connAtoB"
+    ELSE IF chainID = "chainB"
+         THEN "connBtoA"
+         ELSE nullConnectionID      
+
+\* get the connection ID of the connection end at chainID's counterparty chain
+GetCounterpartyConnectionID(chainID) ==
+    IF chainID = "chainA"
+    THEN "connBtoA"
+    ELSE IF chainID = "chainB"
+         THEN "connAtoB"
+         ELSE nullConnectionID      
+
+(***************************************************************************
+ Connection datagram handlers
+ ***************************************************************************)
+ 
+\* Handle "ConnOpenInit" datagrams
+HandleConnOpenInit(chainID, chain, datagrams) ==
+    \* get "ConnOpenInit" datagrams, with a valid connection ID
+    LET connOpenInitDgrs == {dgr \in datagrams : 
+                            /\ dgr.type = "ConnOpenInit"
+                            /\ dgr.connectionID = GetConnectionID(chainID)} IN
+    
+    IF connOpenInitDgrs /= {} 
+    \* if there are valid "ConnOpenInit" datagrams, create a new connection end and update the chain
+    \* TODO: check here if connection is already in INIT?
+    THEN LET connOpenInitDgr == CHOOSE dgr \in connOpenInitDgrs : TRUE IN
+         LET connOpenInitConnectionEnd == [
+             state |-> "INIT",
+             connectionID |-> connOpenInitDgr.connectionID,
+             clientID |-> connOpenInitDgr.clientID,
+             counterpartyConnectionID |-> connOpenInitDgr.counterpartyConnectionID,
+             counterpartyClientID |-> connOpenInitDgr.counterpartyClientID,
+             channelEnd |-> chain.connectionEnd.channelEnd 
+         ] IN 
+         LET connOpenInitChain == [
+             chain EXCEPT !.connectionEnd = connOpenInitConnectionEnd
+         ] IN
+        
+         connOpenInitChain
+    \* otherwise, do not update the chain     
+    ELSE chain
+    
+
+\* Handle "ConnOpenTry" datagrams
+HandleConnOpenTry(chainID, chain, datagrams) ==
+    \* get "ConnOpenTry" datagrams, with a valid connection ID and valid height
+    LET connOpenTryDgrs == {dgr \in datagrams : 
+                            /\ dgr.type = "ConnOpenTry"
+                            /\ dgr.desiredConnectionID = GetConnectionID(chainID)
+                            /\ dgr.consensusHeight <= chain.height
+                            /\ dgr.proofHeight <= chain.counterpartyClientHeight} IN
```

Comment:
I believe we need to have a `chain.counterpartyClientHeights` to simulate the full set of consensus states on this chain and then check that `dgr.proofHeight \in chain.counterpartyClientHeight`